game-core/src/main/java/games/strategy/triplea/ui/screen/TileManager.java

```diff
@@ -371,6 +377,23 @@ private void drawUnits(final Territory territory, final MapData mapData, final S
     }
   }
 
+  private static List<UnitCategory> getSortedUnitCategories(final Territory t, final GameData data) {
+    final List<UnitCategory> categories = new ArrayList<>(UnitSeperator.categorize(t.getUnits().getUnits()));
+    final List<PlayerID> players = data.getPlayerList().getPlayers();
+    if (players.contains(t.getOwner())) {
+      players.remove(t.getOwner());
+      players.add(0, t.getOwner());
```

Comment:
This seems like a smell to me.
Why isn't the comparator method below being used for this?

Reply:
I guess you could embed this code into the comparator but from a readability standpoint, seemed better to sort the player list separately from sorting the category list. If you have a clean way to do it then I'd be interested in taking a look.

Reply:
Haven't tested this, but something along those lines should work (based on my other code snippet):
```java
categories.sort(Comparator
  .comparing(UnitCategory::getOwner, Comparator
    .comparing(t.getOwner()::equals) // Boolean implements Comparable and true is considered greater than false
    .thenComparing(players::indexOf))
  .thenComparing(UnitCategory::getType, Comparator
    .comparing(ut -> !UnitAttachment.get(ut).getIsInfrastructure())
    .thenComparing(ut -> !Matches.unitTypeIsLand().test(ut))
    .thenComparing(ut -> !UnitAttachment.get(ut).getIsSea())
    .thenComparing(xmlUnitTypes::indexOf));
```

Reply:
Updated.